slides/extras-effect-encodings.qmd

```diff
@@ -0,0 +1,356 @@
+  labs(x = "Average ADR per agent")
+```
+:::
+:::
+
+
+## What is an effect encoding?
+
+We replace the qualitative’s predictor data with their _effect on the outcome_. 
+
+::: columns
+::: {.column width="50%"}
+Data before:
+
+```{r}
+#| label: before
+before
+```
+
+:::
+
+::: {.column width="50%"}
+
+Data after:
+
+```{r}
+#| label: after
+after
+```
+
+:::
+:::
+
+The `agent` column is replaced with an estimate of the ADR. 
+
+
+## Per-agent statistics again {.annotation}
+
+::: columns
+::: {.column width="50%"}
+```{r}
+#| label: effects-again 
+#| echo: false
+#| out-width: '90%'
+#| fig-width: 6
+#| fig-height: 3
+#| fig-align: 'center'
+#| dev: 'svg'
+#| dev-args: list(bg = "transparent")
+  
+agent_stats %>%   
+  ggplot(aes(x = num_reservations)) +
+  geom_histogram(bins = 30, col = "blue", fill = "blue", alpha = 1/3) +
+  labs(x = "Number of reservations per agent")
+agent_stats %>%   
+  ggplot(aes(x = ADR)) +
+  geom_histogram(bins = 30, col = "red", fill = "red", alpha = 1/3) +
+  labs(x = "Average ADR per agent")
+```
+:::
+
+::: {.column width="50%"}
+
+- Good statistical methods for estimating these rates use *partial pooling*.
+
+
+- Pooling borrows strength across agents and shrinks extreme values towards the mean for agents with very few transations
+
+
+- The embed package has recipe steps for effect encodings.
+
+:::
+:::
+
+
+:::notes
+Partial pooling gives better estimates for agents with fewer reservations by shrinking the estimate to the overall ADR mean
+
+
+:::
+
+## Partial pooling
+
+```{r}
+#| label: effect-compare 
+#| echo: false
+#| fig-width: 6
+#| fig-height: 6
+#| fig-align: 'center'
+#| dev: 'svglite'
+#| dev-args: list(bg = "transparent")
+
+inner_join(agent_stats, estimates, by = "agent") %>% 
+  ggplot(aes(x = ADR, y = estimate)) + 
+  geom_abline(col = "green", lty = 2) +
+  geom_point(aes(size = num_reservations), alpha = 1/3) +
+  coord_obs_pred() +
+  scale_size(range = c(1/3, 5)) +
+  labs(x = "ADR Sample Mean", y = "Estimated via Effects Encoding")
+```
+
+## Agent effects `r hexes("recipes","embed")` {.annotation}
+
+```{r}
+#| label: lencode-rec
+#| code-line-numbers: "1,6|"
+library(embed)
+
+hotel_effect_rec <-
+  recipe(avg_price_per_room ~ ., data = hotel_tr) %>% 
+  step_YeoJohnson(lead_time) %>%
+  step_lencode_mixed(agent, company, outcome = vars(avg_price_per_room)) %>%
+  step_dummy(all_nominal_predictors()) %>%
+  step_zv(all_predictors())
+```
+
+. . .
+
+It is very important to appropriately validate the effect encoding step to make sure that we are not overfitting.
+
+## Recipes are estimated `r hexes("recipes")`
+
+Preprocessing steps in a recipe use the *training set* to compute quantities.
+
+. . .
+
+What kind of quantities are computed for preprocessing?
+
+-   Levels of a factor
+-   Whether a column has zero variance
+-   Normalization
+-   Feature extraction
+-   Effect encodings
+
+. . .
+
+When a recipe is part of a workflow, this estimation occurs when `fit()` is called.
```

Comment:
this feels duplicate of https://github.com/tidymodels/workshops/pull/127#pullrequestreview-1520784203

Reply:
It does (on purpose) but it is pretty redundant so I'll take it out.